### DISCUSSION TITLE
Conflict between iotdb-session and hive-jdbc

### DISCUSSION THREAD

**Discussion by haowdev** [apache/iotdb#3884] [Q&A] [answered]
Experts, 

We have a maven project with following dependencies in pom.xml.
```
<dependency>
        <groupId>org.apache.iotdb</groupId>
        <artifactId>iotdb-session</artifactId>
        <version>0.12.1</version>
    </dependency>
   <dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>1.1.0</version>
   </dependency>
   <dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-common</artifactId>
            <version>2.7.3</version>
   </dependency>
```

With only adding the dependency of "iotdb-session". the code works well that I can query data from iotdb.

```
<dependency>
        <groupId>org.apache.iotdb</groupId>
        <artifactId>iotdb-session</artifactId>
        <version>0.12.1</version>
    </dependency>
```

However, exception happened when I add following dependencies.

```
   <dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>1.1.0</version>
   </dependency>
   <dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-common</artifactId>
            <version>2.7.3</version>
   </dependency>
```

SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/haow/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/haow/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/thrift/TConfiguration
	at org.apache.iotdb.rpc.TConfigurationConst.<clinit>(TConfigurationConst.java:26)
	at org.apache.iotdb.session.SessionConnection.init(SessionConnection.java:90)
	at org.apache.iotdb.session.SessionConnection.<init>(SessionConnection.java:83)
	at org.apache.iotdb.session.Session.constructSessionConnection(Session.java:291)
	at org.apache.iotdb.session.Session.open(Session.java:261)
	at org.apache.iotdb.session.Session.open(Session.java:246)
	at connectiotdb.main(connectiotdb.java:9)
Caused by: java.lang.ClassNotFoundException: org.apache.thrift.TConfiguration
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 7 more

Process finished with exit code 1

**Comment by haowdev** [comment]
Any update on this incompatible issue?

**Comment by qiaojialin** [comment, marked as answer]
This is due to thrift dependency conflict between iotdb-session and hive.

Try to use ```mvn dependency:tree``` to see which dependency conflicts with iotdb-session.

Then, use <exclusion> to remove the dependency

```
<dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>1.1.0</version>
            <exclusions>
                <exclusion>
                    <groupId>org.apache.thrift</groupId>
                    <artifactId>libthrift</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```
